src/patternfly/components/Table/table-grid.scss

```diff
@@ -172,12 +172,12 @@
   }
 
   // Thead
-  thead {
+  > thead {
     display: none;
   }
```

Comment:
Not sure if this is by design or not, but all nested tables currently take on a responsive/grid layout if they're in a table with a responsive/grid layout, even when the nested table doesn't have a `.pf-m-grid*` class. Limiting the scope of selectors like this breaks this selector's role a nested table's responsive layout. For example, in this compound expansion example (https://patternfly-pr-5558.surge.sh/components/table#compound-expansion), the `thead` stuff is no longer hidden:

<img width="619" alt="Screenshot 2023-05-15 at 8 26 32 PM" src="https://github.com/patternfly/patternfly/assets/35148959/97d16179-8842-4c2f-b873-a25569d90a2d">

I wonder if maybe there are a few things we can do:
1. Add a BEM class for all things - don't use child/sibling selectors as a means to tie an element to a version. That should allow for a 1:1 replacement? So for this selector, update `thead` to `thead:where(.pf-c-table__thead)` and don't use `>`. IMO that would be the easiest, and most consistent. It would also be more clear if everything had a class, since we have a mix/match of this kind of thing`tr:where($table__tr)` and `$table > tr` and it isn't super obvious as to why. I'm often having to look at the compiled CSS to track down whether a style is tied to a versioned selector.
2. Try to fix these selectors so they behave the way they did before. I think this selector would either 1) need a `thead:where($table__thead)`? Which I guess makes me think #1 might just be easier.
3. Try and limit the scope of responsive styles so nested tables aren't impacted. IMO that would be a good change regardless. Right now you can't have a non-responsive table layout nested in a responsive layout parent table. 

Other ideas or thoughts?